tests/test_execution_client.cpp

```diff
@@ -0,0 +1,28 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "sm_tests.h"
+
+namespace ara::exec {
+    /* Mock ExecutionClient for unit testing. */
+    class MockExecutionClient : public ExecutionClient {
+    public:
+        MOCK_METHOD(void, ReportApplicationState, (sm::FunctionGroupStateType state));
+    };
+}
+
+TEST_F(smTests,  testExecutionClient)
+{
+    ara::exec::MockExecutionClient mockEC;
+    mySM.executionClient = &mockEC;
+    EXPECT_CALL(mockEC,
+                ReportApplicationState(ara::sm::FunctionGroupStateType::On)).Times(testing::AnyNumber());
+
+    mySM.stateClient->SmSetState(ara::sm::FunctionGroupStateType::On);
```

Comment:
You can also add test for report ApplicationState off

Reply:
Done